Tests/SwinjectTests/NestedAssemblers.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright © 2021 Swinject Contributors. All rights reserved.
+//
+
+import Swinject
+
+struct NestedAssembly: Assembly {
+
+    typealias ArrayType = NSMutableArray
```

Comment:
The testing concept injects two (reference type) arrays to allow tracking the order of the assemble and load operations. Seems easiest to do with `NSMutableArray`. Just first thing that came to mind